src/lib/trust_system/trust_system.ml

```diff
@@ -26,6 +52,28 @@ module Actions = struct
     (* FIXME figure out a good value for this *)
     let request_increment = Peer_trust.max_rate 10. in
     match action with
+    | Gossiped_old_transition slot_diff ->
+        (* We want to decrease the score exponentially based on how out of date the transition
```

Comment:
Two points:

* According to Corey, honest peers gossip transitions that are after the current slot but before the end of the current slot plus delta. So we shouldn't punish that behavior. In addition, due to network latency and unavoidable clock skew, peers will slightly miss that cutoff. We should soft-punish any incoming transitions that are more than delta slots too old. Since all out of date transitions cost us the same amount to process I think the punishment value should just be a constant, for simplicity's sake.

* This is actually geometric and not exponential. The exponential fit is `y = 2/(e^Δ) * e^x`. There's not enough degrees of freedom in a exponential of the form `y = a * e^(b*x)` to let you set `f(1)` freely. It's `2 * e(-Δ)` in the equation I gave, which works out to 0.0996 for Δ = 3, and 0.005 for Δ = 6. This is irrelevant if we're just using a constant though.

Reply:
> honest peers gossip transitions that are after the current slot but before the end of the current slot plus delta

I think this is a bug ^^

We'll have to fix the performance problem or compensate by sufficiently increasing the delta in order to get the guarantees of ouroboros to be happy.

Reply:
Discussed out of band -- there was some confusion over the definition of the cutoff time for which the `slot_diff` is based on. I will update the comment to define this more clearly. `slot_diff` is `received_slot - (proposed_slot + Δ)`.

Reply:
> > honest peers gossip transitions that are after the current slot but before the end of the current slot plus delta
> 
> I think this is a bug ^^
> 
> We'll have to fix the performance problem or compensate by sufficiently increasing the delta in order to get the guarantees of ouroboros to be happy.

Are you saying that Ouroboros disallows rebroadcasting of blocks outside of their proposed slot entirely? I thought that network delay was not necessarily just a calculation of the direct network communication delay, but also accounted for network hopping via rebroadcasting.

Reply:
I think I'm going to keep the geometric implementation so that we can have easier fine grained control over the soft punishment for the `slot_diff == 1` since I think tuning that will actually be important. I will update the comment to replace the incorrect adjective I used.